core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -507,4 +513,26 @@ protected static Fuzziness randomFuzziness(String fieldName) {
     protected static boolean isNumericFieldName(String fieldName) {
         return INT_FIELD_NAME.equals(fieldName) || DOUBLE_FIELD_NAME.equals(fieldName);
     }
+
+    private static class MockTermsLookupFetchService extends TermsLookupFetchService {
```

Comment:
Don't want to digress, but this would be a perfect usecase for Mockito. Just saying, as long as this works. Might be something to look into later.